tests/toplevel_unit.c

```diff
@@ -57,6 +57,45 @@ DEFINE_TEST(range_contains) {
     roaring_bitmap_free(bm);
 }
 
+DEFINE_TEST(contains_multi) {
+    // create rbm with RLE conatainer form 0 1000
+    roaring_bitmap_t *bm = roaring_bitmap_from_range(0, 1001, 1);
+    // add array container from 66000
+    for (uint32_t i = 77000; i < 87000; i+=2) {
+        roaring_bitmap_add(bm, i);
+    }
+    // add bitset container from 20000
+    for (uint32_t i = 132000; i < 140000; i+=2) {
+        roaring_bitmap_add(bm, i);
+    }
+
+    size_t length = 10;
+    bool results[length];
```

Comment:
This is not correct strictly speaking. 
Try:

```
const size_t length = 10;
bool results[length];
```

Reply:
fixed
My compiler did not even WARNING me about this.

Reply:
I expect that @Dr-Emann is correct. In any case, the code needs fixing.

Reply:
> > Why it is not recommended to use stack memory?
> 
> It is _recommended_ to prefer stack when possible, but stack memory is limited... in some context it can be relatively scarce. If the stack memory is insufficient, you get an unsafe crash.
> 
> The following is definitively fine:
> 
> ```
> bool results[10];
> ```
> 
> Please don't do...
> 
> ```
> bool *results = (bool *)malloc(10 * sizeof(bool));
> ```
> 
> In the case where you do not know the size of the array (e.g., it could be 10, 1000, 1000000), then you cannot use stack allocation.


I get it, thanks a lot

Reply:
It's not a question of suggested, or not suggested. In C without VLAs, you _cannot_ use the stack for an allocation which does not have a known size at compile time. `argc` is the count of the arguments passed to the executable: it _cannot_ be known until the code is running, and therefore _cannot_ be used as the size of an array.

Your options are:
* Pick an upper limit, and use that size:
    `if (argc > 100) abort(); uint32_t *values[100]; ....`
* Allocate a dynamic size using malloc:
    `uint32_t **values = malloc(argc * sizeof(*values));`

You can pass `-Werror=vla` to gcc to get errors locally to validate everything is correct (`cmake -DCMAKE_C_FLAGS="-Werror=vla" -DCMAKE_CXX_FLAGS="-Werror=vla" .`

@lemire if we want to support msvc, it might be worth adding `-Werror=vla` to the compile flags, so we get early errors when building locally outside of msvc?

Reply:
> it might be worth adding

PR invited but note that it is not a huge problem in practice.